src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const firstState = { ...state };
+  const changeState = [];
+
+  function assigned(index, to, from) {
+    changeState[index] = Object.assign(to, from);
+  }
+
+  for (let i = 0; i < actions.length; i++) {
+    if (actions[i].type === 'addProperties') {
+      for (const key in actions[i].extraData) {
+        firstState[key] = actions[i].extraData[key];
+      }
+      assigned(i, {}, firstState);
```

Comment:
Don't repeat yourself: you use this functionality 3 times in your code, move it at the end of cycle